src/popup/pages/MultisigProposalDetails.vue

```diff
@@ -0,0 +1,611 @@
+import DialogBox from '../components/DialogBox.vue';
+import MultisigProposalConsensus from '../components/MultisigProposalConsensus.vue';
+import Avatar from '../components/Avatar.vue';
+import TransactionTokens from '../components/TransactionTokenRows.vue';
+
+import AnimatedSpinner from '../../icons/animated-spinner.svg?skip-optimize';
+import ExternalLink from '../../icons/external-link.svg?vue-component';
+import { ROUTE_ACCOUNT } from '../router/routeNames';
+
+export default defineComponent({
+  components: {
+    PayloadDetails,
+    TransactionTokens,
+    MultisigProposalConsensus,
+    TransactionInfo,
+    DialogBox,
+    Avatar,
+    TokenAmount,
+    DetailsItem,
+    LinkButton,
+    AccountItem,
+    BtnMain,
+    CopyText,
+    AnimatedSpinner,
+    ExternalLink,
+  },
+  setup(props, { root }) {
+    const {
+      activeMultisigAccount,
+      updateMultisigAccounts,
+    } = useMultisigAccounts({ store: root.$store });
+
+    const {
+      pendingMultisigTxExpired,
+      pendingMultisigTxExpiresAt,
+      pendingMultisigTxCanBeSent,
+      pendingMultisigTxLocalSigners,
+      pendingMultisigTxConfirmedByLocalSigners,
+    } = usePendingMultisigTransaction({ store: root.$store });
+
+    const {
+      fetchActiveMultisigTx,
+      fetchTransactionByHash,
+      sendTx,
+      callContractMethod,
+    } = useMultisigTransactions({
+      store: root.$store,
+    });
+
+    const {
+      isLocalAccountAddress,
+    } = useAccounts({
+      store: root.$store,
+    });
+
+    const getExplorerPath = useGetter('getExplorerPath');
+    const isTxAex9 = useGetter('isTxAex9');
+    const getTxSymbol = useGetter('getTxSymbol');
+    const getTxAmountTotal = useGetter('getTxAmountTotal');
+
+    const processingAction = ref<boolean>(false);
+    const multisigTx = ref<IActiveMultisigTx | null>(null);
+    const transaction = ref<ITransaction | null>(null);
+    const proposalCompleted = ref<boolean>(false);
+
+    const totalSpent = computed(() => {
+      if (!proposalCompleted.value || !transaction.value) return 0;
+      const { tx } = transaction.value;
+      const multisigTransaction = tx.tx?.tx;
+      if (!multisigTransaction) return 0;
+
+      return (
+        +(multisigTransaction.amount) + tx.fee + multisigTransaction.fee
+      );
+    });
+
+    async function getTransactionDetails(currentAccount: IMultisigAccount) {
+      const activeMultisigTx = await fetchActiveMultisigTx(currentAccount.contractId);
+
+      if (!activeMultisigTx) {
+        return;
+      }
+
+      multisigTx.value = {
+        ...activeMultisigTx.tx,
+        type: SCHEMA.TX_TYPE.gaMeta,
+        tag: SCHEMA.TX_TYPE.gaMeta,
+      } as any;
+    }
+
+    /**
+     * Utilized to open the confirmation modal for approving, disapproving, or revoking the proposal
+     */
+    async function dispatchProposalAction(action: IMultisigFunctionTypes) {
+      if (!activeMultisigAccount.value) return;
```

Comment:
for better readability I suggest to wrap `return` with braces
```suggestion
      if (!activeMultisigAccount.value) {
        return;
      }
```